[RN] (Feature) Export Screen - Add Wallet address + update the design + add warning popup

serdiukov-o-nordwhale opened the pull request:
# Description

- added Wallet address
- added warning popup
- updated  design
- adjusted unit tests

About #2597 

# Screenshots
<img width="464" alt="image" src="https://user-images.githubusercontent.com/62208067/97315556-84019780-1871-11eb-83d0-777c2fcf84e3.png">
<img width="473" alt="image" src="https://user-images.githubusercontent.com/62208067/97315622-91b71d00-1871-11eb-94b8-74f006826289.png">

<img width="473" alt="image" src="https://user-images.githubusercontent.com/62208067/97315878-d478f500-1871-11eb-95ec-559bb8e57159.gif">

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [x] I have added screenshots related to my pull request (for frontend tasks)
- [x] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

LiavGut commented:
@serdiukov-o-nordwhale The avatar photos should be a circle (as per design)

serdiukov-o-nordwhale commented:
> @serdiukov-o-nordwhale The avatar photos should be a circle (as per design)

They are circles, i changed nothing for their styles, have took existing desing:

```
width: getDesignRelativeWidth(imageSize, false),
borderRadius: getDesignRelativeHeight(halfImageSize, false), // half of height/width
```

squared box with half-width border radius will be displayed as the circle

mauriguz commented on src/components/backupWallet/ExportWalletData.js:
```diff
@@ -24,8 +25,15 @@ import config from '../../config/config'
+import ExportWarningPopup from './ExportWarningPopup'
 
-const web3ProviderUrl = GoodWallet.networkId && config.ethereum[GoodWallet.networkId].httpWeb3provider
+if (Platform.OS === 'web') {
```
this could be removed - prefetching svgs in react-native creates uncaught promise errors, we have had some of these in the past

mauriguz commented on src/components/backupWallet/ExportWarningPopup.js:
```diff
@@ -0,0 +1,72 @@
+  },
+  image: {
+    color: theme.colors.primary,
+    transform: 'rotate(180deg)',
```
```suggestion

```
this silently crashes because of applying transform to text. instead it could be applied to imageWrapper and the result is the same but without the error

mauriguz commented on src/components/backupWallet/ExportWarningPopup.js:
```diff
@@ -0,0 +1,72 @@
+  imageWrapper: {
+    display: 'flex',
+    alignItems: 'center',
+  },
```
```suggestion
    transform: [{ rotate: '180deg' }],
  },
```

mauriguz commented:
@serdiukov-o-nordwhale in native I'm also seeing another thing, when I get into the export screen I see my private key but no wallet adress nor fuse url. after forcing a hot refresh doing some change in the code they both appear and all information is available.
I don't see anything in this PR that could be causing that so I'm thinking this error is already part of this branch when getting Goodwallet and seems to be outside of the scope for this PR, though, let me know if you'd like to fix it now, otherwise we can create an issue to fix it later

sirpy reviewed (changes requested):
1. the non round avatar could be because of the use of getRelative... try to give the borderRadius also the image size and not half
2. @mauriguz what do you mean? it doest show but then shows? are there errors? does it always happen? please share a video

sirpy commented on src/components/backupWallet/ExportWalletData.js:
```diff
@@ -35,14 +43,25 @@ type ExportWalletProps = {
-  const handleGoHome = useOnPress(() => navigate('Home'), [navigate])
+  const fullPrivateKey = useMemo(() => get(GoodWallet, 'wallet.eth.accounts.wallet[0].privateKey', ''), [])
+
+  const onPrivateKeyCopied = useCallback(() => {
```
```suggestion
  const onPublicKeyCopied = useCallback(() => {
```

sirpy commented on src/components/backupWallet/ExportWalletData.js:
```diff
@@ -52,22 +71,37 @@ const ExportWalletData = ({ navigation, styles, theme }: ExportWalletProps) => {
+            imageSource={avatarSource}
+            copyButtonText="Copy address"
+            showCopyIcon={false}
+            onCopied={onPrivateKeyCopied}
```
```suggestion
            onCopied={onPublicKeyCopied}
```

serdiukov-o-nordwhale commented:
> * the non round avatar could be because of the use of getRelative... try to give the borderRadius also the image size and not half

i remember that i saw the styles like:
```
width: getRelativeDesign**Width**()....,
height: getRelativeDesign**Width**()...., // relative width was used fot height too
```

I wondered why getRelativeDesign_Width_ is used to calculate view _height_, and i fixed that. Now i've understood - that's a way so make images the exact circles. will restore those styles back

serdiukov-o-nordwhale commented:
I added suggestions. Regarding the strange public key / fuse rpc behavior - i moved getting those values inside the component. @mauriguz please re check does the issue still exists or it was fixed

mauriguz commented:
> I added suggestions. Regarding the strange public key / fuse rpc behavior - i moved getting those values inside the component. @mauriguz please re check does the issue still exists or it was fixed

@serdiukov-o-nordwhale @sirpy fix works and now all the information is available!

mauriguz commented on src/components/backupWallet/ExportWalletData.js:
```diff
@@ -52,22 +69,39 @@ const ExportWalletData = ({ navigation, styles, theme }: ExportWalletProps) => {
             title="Fuse Network RPC Address"
             content={web3ProviderUrl}
+            imageSize={50}
             imageSource={FuseLogo}
```
```suggestion
            image={FuseLogo}
```
